### PR TITLE
Feature - 6529 - iap welcome step copy

### LINKED ISSUE #6529: ✨ Update Welcome step copy for IAP Application
## ✨ Feature

Change the text of the welcome step slightly for IAP Applications.

## 🕵️ Details

Add any additional details that could assist with the development of the new feature.

## 🎨 Design File

https://www.figma.com/file/8piNhFaBjcxssBB8LAGxEY/IAP-User-flows?type=design&node-id=269-76379&t=fgKSYtT4BVSNN2CW-0

## 📸 Screenshot

Add a screenshot of the new feature (if possible).

## 🙋‍♀️ Proposed Implementation

Just change the copy slightly for IAP Applications

## 🌎 Localization

(optional) Provide any new copy along with translations available.

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the feature was properly implemented.

- [ ] Text on Welcome step for IAP Application matches figma design

### PR DESCRIPTION
**Pull request by yonikid15**
🤖 Resolves #6529 

## 👋 Introduction

- Change the text of the welcome step slightly for IAP Applications.

## 🕵️ Details

I wasn't sure which route to take so I decided to post both (one is commented out) and see what others think is best.

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

1. Create an application in digital talent.
2. Go to welcome step and copy and ensure it's correct.
3. Change pool advertisement publishing group to `IAP` in Database.
4. Go to application welcome step again and ensure some of the copy has changed.

### CODE REVIEW

**Comment by vd1992** [comment]
I like that approach, the welcome page especially feels like it could be something you really want modular and so would be good to be compatible with more than two possible messages. Ternary, method 2, stops working once one more message case appears. 

Eric's PR (#6521) for this introduced a context component, not sure whether you ought to use it. That might be where all further application context is controlled with any new features/conditions, so might be good to use instead

https://github.com/GCTC-NTGC/gc-digital-talent/pull/6521/files#diff-45ca1059a280f3190c7a0daaa9022e5b8eed7a42632df28ae650e74ea57710f0

File: apps/web/src/pages/Applications/ApplicationWelcomePage/ApplicationWelcomePage.tsx [hunk 1]
```diff
@@ -113,13 +119,23 @@ const ApplicationWelcome = ({ application }: ApplicationPageProps) => {
...
+              description:
+                "Description of how the skills-based hiring platform assess candidates.",
+            })
+          : intl.formatMessage({
+              defaultMessage:
+                "The GC Digital Talent platform is a skills-based hiring system. This means that your application will put a heavier focus on your skills and how you've used them in past experiences to help us get a stronger understanding of your fit.",
+              id: "u/DBSl",
+              description:
+                "Description of how the skills-based hiring platform assess candidates.",
+            })} */}
```
Comment (esizer) [inline apps/web/src/pages/Applications/ApplicationWelcomePage/ApplicationWelcomePage.tsx:138]: I think this method would be preferred for now until we need different text, then we could easily convert to the second method with some tweaks.

Also, we introduced a context provider for this exact thing. You can do the following and it should work:

```tsx
const { isIAP } = useApplicationContext();
```

Comment (yonikid15) [reply apps/web/src/pages/Applications/ApplicationWelcomePage/ApplicationWelcomePage.tsx:138]: Okay, I'll you this method for now until we decide to add more copy. Thanks!

File: apps/web/src/pages/Applications/ApplicationWelcomePage/ApplicationWelcomePage.tsx [hunk 2]
```diff
@@ -13,6 +13,8 @@ import { useUpdateApplicationMutation, ApplicationStep } from "~/api/generated";
 
 import { errorMessages } from "@gc-digital-talent/i18n";
 import ApplicationApi, { ApplicationPageProps } from "../ApplicationApi";
+import { PublishingGroup } from "@gc-digital-talent/graphql";
```
Comment (esizer) [inline apps/web/src/pages/Applications/ApplicationWelcomePage/ApplicationWelcomePage.tsx:16]: This can be removed now

**Review by esizer — CHANGES_REQUESTED** [review]
Looks good! Just one more import to be removed 😛

File: apps/web/src/pages/Applications/ApplicationWelcomePage/ApplicationWelcomePage.tsx (hunk 2 above)
Comment (yonikid15) [reply apps/web/src/pages/Applications/ApplicationWelcomePage/ApplicationWelcomePage.tsx]: 😅 Fixed by [5d9ed81](https://github.com/GCTC-NTGC/gc-digital-talent/pull/6545/commits/5d9ed81013041c5c62f0f228a38bc42cf21de005)

**Comment by yonikid15** [comment]
Just waiting on translations now. I paired it up with the translations from https://github.com/GCTC-NTGC/gc-digital-talent/pull/6541, which is why it's taking a bit longer.

**Review by esizer — APPROVED** [review]

File: apps/web/src/pages/Applications/ApplicationWelcomePage/ApplicationWelcomePage.tsx [hunk 3]
```diff
@@ -118,13 +119,21 @@ const ApplicationWelcome = ({ application }: ApplicationPageProps) => {
...
-          defaultMessage:
-            "The GC Digital Talent platform is a skills-based hiring system. This means that your application will put a heavier focus on your skills and how you've used them in past experiences to help us get a stronger understanding of your fit.",
-          id: "u/DBSl",
-          description:
-            "Description of how the skills-based hiring platform assess candidates.",
-        })}
+        {isIAP
+          ? intl.formatMessage({
+              defaultMessage:
+                "The Program is a Government of Canada initiative specifically for First Nations, Inuit, and Métis peoples. It is pathway to employment in the federal public service for Indigenous peoples who have a passion for Information Technology (IT). We focus on that passion, and your potential to grow and succeed in this field.",
```
Comment (mnigh) [inline apps/web/src/pages/Applications/ApplicationWelcomePage/ApplicationWelcomePage.tsx:125]: - word **program** is not a proper noun and should be lowercase
- second sentence missing the indefinite article **a** before pathway

```suggestion
                "The program is a Government of Canada initiative specifically for First Nations, Inuit, and Métis peoples. It is a pathway to employment in the federal public service for Indigenous peoples who have a passion for Information Technology (IT). We focus on that passion, and your potential to grow and succeed in this field.",
```

@substrae please update the [figma](https://www.figma.com/file/8piNhFaBjcxssBB8LAGxEY/IAP-User-flows?type=design&node-id=269-76293&t=7ETAEU0FCLxI8awi-0) file accordingly.

File: apps/web/src/pages/Applications/ApplicationWelcomePage/ApplicationWelcomePage.tsx [hunk 4]
```diff
@@ -118,13 +119,21 @@ const ApplicationWelcome = ({ application }: ApplicationPageProps) => {
...
-          description:
-            "Description of how the skills-based hiring platform assess candidates.",
-        })}
+        {isIAP
+          ? intl.formatMessage({
+              defaultMessage:
+                "The program is a Government of Canada initiative specifically for First Nations, Inuit, and Métis peoples. It is a pathway to employment in the federal public service for Indigenous peoples who have a passion for Information Technology (IT). We focus on that passion, and your potential to grow and succeed in this field.",
+              id: "rJLUje",
+              description:
+                "Description of how the skills-based hiring platform assess candidates.",
```
Comment (mnigh) [inline apps/web/src/pages/Applications/ApplicationWelcomePage/ApplicationWelcomePage.tsx:128]: This description is exactly the same as another string that is not the same. Don't forget to update the id and description in fr.json as this change will modify its id and description.
```suggestion
                "Description of how the hiring platform assesses candidates for IAP.",
```

File: apps/web/src/lang/fr.json [hunk 5]
```diff
@@ -8152,6 +8152,10 @@
     "defaultMessage": "Le passage réussi de deux années d’éducation postsecondaire en sciences informatiques, en technologie de l’information, en gestion de l’information ou dans un autre domaine spécialisé pertinent à ce poste.",
     "description": "post secondary education experience for pool advertisement"
   },
+  "VHhOb/": {
+    "defaultMessage": "Le programme est une initiative du gouvernement du Canada, qui est conçue spécifiquement pour les Premières Nations, les Inuits et les Métis. Il s’agit d’un chemin vers l’emploi au sein de la fonction publique fédérale pour les Autochtones qui sont passionnés de technologies de l’information (TI). Nous misons sur cette passion ainsi que sur votre potentiel pour grandir et réussir dans ce domaine.",
+    "description": "Description of how the skills-based hiring platform assess candidates."
```
Comment (mnigh) [inline apps/web/src/lang/fr.json:8157]:
```suggestion
    "description": "Description of how the hiring platform assesses candidates for IAP."
```

**Review by mnigh — CHANGES_REQUESTED** [review]

**Pull request merged**